app/controllers/api/map_controller.rb

```diff
@@ -5,6 +5,13 @@ class MapController < ApiController
     before_action :check_api_readable
     around_action :api_call_handle_error, :api_call_timeout
 
+    before_action :default_format_xml
+
+    # Set format to xml unless client requires a specific format
+    def default_format_xml
```

Comment:
There's no need to repeat the function definition in each controller. You can move the definition to `app/controllers/api_controller.rb`, since all the controllers in `app/controllers/api/` are subclassed from that one.

Reply:
Yup, moved that one to the api_controller for now.